plutus-conformance/add-test-output/Spec.hs

```diff
@@ -61,58 +61,76 @@ runnerReader inp =
   Left ("Unsupported test " <> show inp <>
         ". Please choose either eval (for evaluation tests) or typecheck (for typechecking tests).")
 
+data SomeOrAll =
+  Missing
+  | All
+
+missing :: Parser SomeOrAll
+missing =
+  flag' Missing ( long "missing"
+    <> short 'm'
+    <> help "only add missing outputs" )
+
+allInputs :: Parser SomeOrAll
+allInputs =
+  flag' All ( long "all"
+    <> short 'a'
+    <> help "add outputs to all input files" )
+
+allOrMissing :: Parser SomeOrAll
+allOrMissing = missing <|> allInputs
+
 args :: ParserInfo Args
-args = info ((MkArgs <$> ext <*> dir <*> runner) <**> helper)
+args = info ((MkArgs <$> ext <*> dir <*> runner <*> allOrMissing) <**> helper)
   (fullDesc <> progDesc helpText)
 
 helpText :: String
 helpText = unlines
   ["This program adds test outputs to specified inputs."
-  , "To run the program, input the following 3 arguments: "
+  , "To run the program, input the following 4 arguments: "
   , "(1) file extension to be searched "
   , "(2) directory to be searched "
   , "(3) the action to run the input files through; eval (for evaluation tests) or typecheck (for typechecking tests). "
+  , "(4) whether to write output files to all inputs or only the ones missing output files."
   , "E.g. run "
-  , "`cabal run add-test-output .uplc plutus-conformance/uplc/ eval` "
+  , "`cabal run add-test-output .uplc plutus-conformance/uplc/ eval` -- --missing"
   , "to have the executable search for files with extension `.uplc` in the /uplc directory that are missing output files. "
   , " It will evaluate and create output files for them."
   ]
 
-shownEvaluationFailure :: String
-shownEvaluationFailure = "evaluation failure"
-
 main :: IO ()
 main = do
-    MkArgs extension directory run <- customExecParser (prefs showHelpOnEmpty) args
+    MkArgs extension directory run fileOpt <- customExecParser (prefs showHelpOnEmpty) args
     allInputFiles <- findByExtension [extension] directory
-    -- only choose the ones without an output file, so as to not edit the ones already with outputs
-    inputFiles <- filterM (fmap (fmap not) (\testIn -> doesFileExist (testIn <> ".expected"))) allInputFiles
+    inputFiles <-
+      case fileOpt of
+        Missing -> filterM (fmap (fmap not) (\testIn -> doesFileExist (testIn <> ".expected"))) allInputFiles
```

Comment:
Is this using the `Functor` instance for `(->) r` for some reason? Is it just

```haskell
Missing -> filterM (\testIn -> not <$> doesFileExist (testIn <> ".expected")) allInputFiles
```

or is my internal type checker failing me?

Reply:
No. You're right this reads a lot better.